pyhf/pdf.py

```diff
@@ -35,10 +38,13 @@ def from_spec(cls,spec,poiname = 'mu', qualify_names = False):
                     modifier = instance.add_or_get_modifier(channel, sample, modifier_def)
                     modifier.add_sample(channel, sample, modifier_def)
                     modifiers.append(modifier_def['name'])
+                    modifiers_type[modifier_def['name']] = modifier_def['type']
                     sample['modifiers_by_type'].setdefault(modifier_def['type'],[]).append(modifier_def['name'])
         instance.channels = list(set(channels))
         instance.samples = list(set(samples))
         instance.modifiers = list(set(modifiers))
+        instance.modifiers_type = modifiers_type
+        instance.channel_nbins = channel_nbins
         instance.set_poi(poiname)
```

Comment:
I wonder if we should take the opportunity to move all the attribute-assigning things into the `__init__` of `Model` ?

Reply:
I'm not fully convinced that it belongs in `Model` because it's a private member variable. There's no reason for anyone to publicly access this -- which is why I think we agreed to set it on the underlying model config.

Reply:
sorry I did not mean `Model` but config.. e.g. I was thinking we would pass these into init

```
instance = cls(channels samples, modifiers, poiname)
```

etc